In [3]:
import pandas as pd
import seaborn as sns
import matplotlib.pyplot as plt 
sns.set()

In [4]:
%matplotlib inline

In [5]:
df_train = pd.read_csv("../data/train_users_2.csv")
df_test = pd.read_csv("../data/test_users.csv")
df_session = pd.read_csv("../data/sessions.csv")

In [6]:
df_submit = pd.read_csv("../data/sample_submission_NDF.csv")

In [11]:
set_train = set(df_train["id"].unique())
set_test = set(df_test["id"].unique())
set_session = set(df_session["user_id"].unique())

### 중복된 아이디 있는 지 확인

#### 1. train

In [20]:
len(df_train), len(df_train["id"].unique())

(213451, 213451)

#### 2. test

In [21]:
len(df_test), len(df_test["id"].unique())

(62096, 62096)

#### 3. submit

In [15]:
len(df_submit), len(df_submit["id"].unique())

(62096, 62096)

#### 4. session

In [15]:
len(df_session), len(set_session)

(10567737, 135484)

총 id 갯수는 135484개 이다. 

### sesion 내부 test와 train가 있는 지 확인

In [13]:
len(set_train), len(set_test), len(set_session)

(213451, 62096, 135484)

#### 1. session + train

In [20]:
len(set_train & set_session), len(set_session - set_train) 

(73815, 61669)

- train data 안에 id가 session에 있는 갯수 : 73815
- train data 안에 id가 session에 없는 갯수 : 61669

#### 2. session + test

In [22]:
len(set_test & set_session), len(set_session - set_test) 

(61668, 73816)

- test data 안에 id가 session에 있는 갯수 : 61668
- test data 안에 id가 session에 없는 갯수 : 73816

#### 3. session

In [24]:
len(set_session - (set_test | set_train))

1

- test와 train data 안에 id이외 session이 있는 갯수 : 1